### PR TITLE
Update container docu

### PR DESCRIPTION
**Pull request by achimweigel**
**How to categorize this PR?**

/area deployers
/kind enhancement
/priority normal

**What this PR does / why we need it**:

Add link to tutorial for container deplpyer docu.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc user

```

### CODE REVIEW

File: docs/deployer/container.md [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
-The container deployer is a controller that reconciles DeployItems of type `landscaper.gardener.cloud/container`. It executes the given container spec with the injected imports and collect exports from a predefined path.
+The container deployer is a controller that allows you to run whatever coding you need to set up your cloud environment.
+You just need to provide your program as a container image which is then executed by the landscaper in combination
+with the container deployer.
+
+Your image is provided as a reference in a DeployItem of type `landscaper.gardener.cloud/container`. The DeployItem 
+itself must be embedded in a Blueprint and specifies the input data required by your program. The landscaper in
+combination with the container deployer provides these input data during runtime. The output or export data of
+your program must be stored in a predefined file and are collected by the landscaper/container deployer.
```
Comment (MrBatschner) [inline docs/deployer/container.md:10]: ... once your program and thus the container terminates.

**Review by MrBatschner — COMMENTED** [review]
Thank you Achim, makes the documentation much easier to read.
Can you look where "the landscaper" is used a a reference to the whole concept/product and replace it with just "Landscaper"?

File: docs/deployer/container.md (hunk 1 above)
Comment (achimweigel) [reply docs/deployer/container.md]: Hi, I made the requested changes.

File: docs/deployer/container.md [hunk 2]
```diff
@@ -44,14 +55,19 @@ spec:
...
+  corresponding DeployItem was deleted and some optional cleanup could be done.
+- *Imports* are provided as a json file at the path given by the env var `IMPORTS_PATH`.
 - *Exports* should be written to a json or yaml file at the path given by the env var `EXPORTS_PATH`.
-- The optional *state* should be written to the directory given by the env var `STATE_PATH`.
-- The complete state directory will be tarred and managed by the landscaper(:warning: no symlinks)
-- The *Component Descriptor* can be expected as a json file at the path given by the env var `COMPONENT_DESCRIPTOR_PATH`. The json file contains a resolved component descriptor list which means that all transitive component descriptors are included in a list.
+- An optional *state* should be written to the directory given by the env var `STATE_PATH`. The complete state 
+  directory will be tarred and managed by Landscaper(:warning: no symlinks). The last state data are provided 
+  in the next execution or your program. 
+- The *Component Descriptor* can be expected as a json file at the path given by the env var `COMPONENT_DESCRIPTOR_PATH`.
```
Comment (schrodit) [inline docs/deployer/container.md:68]: we should add that this path is only defined if the provider configuration contains a `componentDescriptor` reference

File: docs/deployer/container.md [hunk 3]
```diff
@@ -70,11 +86,16 @@ In order for the container deployer to interact with the landscaper a contract f
   }
   ```
 
-- The optional *content blob* that can be defined by a definition can be accessed at the directory given by the env var `CONTENT_PATH`.
+- The optional *content blob* that can be defined by a definition can be accessed at the directory given by the env
```
Comment (schrodit) [inline docs/deployer/container.md:89]: we should add that the content is only given if a `blueprint` in the provider configuration is defined

File: docs/deployer/container.md [hunk 4]
```diff
@@ -96,6 +117,9 @@ status:
 
 ### Operations
 
-In addition to the annotations that are specified by the deploy item contract (operation reconcile and force-reconcile), the container deployer implements in addition specific annotations that can be set to instruct the container deployer to perform specific actions.
+In addition to the annotations that are specified in the DeployItem contract (operation reconcile and force-reconcile), 
+the container deployer implements in addition specific annotations that can be set to instruct the container deployer to
```
Comment (schrodit) [inline docs/deployer/container.md:121]:
```suggestion
the container deployer implements specific annotations that can be set to instruct the container deployer to 
```
2 times `in addition`

**Review by schrodit — CHANGES_REQUESTED** [review]

File: docs/deployer/container.md (hunk 2 above)
Comment (achimweigel) [reply docs/deployer/container.md]: We have changed this but it was not our goal to do a complete rework of this documentation now. We think this document is still not in a good state. It is even unclear what is the target audience. We plan to do a complete rework later.

**Review by schrodit — APPROVED** [review]

**Comment by schrodit** [comment]
@MrBatschner can have another look?

**Pull request merged**